test/Interop/Cxx/namespace/class-inline-namespace-irgen.swift

```diff
@@ -0,0 +1,125 @@
+// RUN: rm -rf %t
+// RUN: split-file %s %t
+// RUN: %target-swift-emit-ir -I %t/Inputs -enable-cxx-interop %t/test.swift | %FileCheck %t/test.swift
+
+
+//--- Inputs/module.modulemap
+module namespaces {
+  header "test.h"
+  requires cplusplus
+}
+//--- Inputs/test.h
+namespace Parent {
+inline namespace InlineChild {
+
+void functionInInlineChild();
+
+template<class CharT>
+struct TemplateInInlineChild {
+};
+
+typedef TemplateInInlineChild<char> TypedefInInlineChild;
+
+struct InInlineChild {
+};
+
+namespace NamespaceInInlineChild {
+
+struct InNamespaceInInlineChild {
+};
+
+} // namespace NamespaceInInlineChild
+
+inline namespace SecondInlineChild {
+
+struct InSecondInlineChild {
+};
+
+} // namespace SecondInlineChild
+} // namespace InlineChild
+} // namespace Parent
+
+//--- test.swift
+
+import namespaces;
+
+extension Parent.TypedefInInlineChild {
+  var string: String {
+    return ""
+  }
+}
+// CHECK: define hidden swiftcc { i64, %swift.bridge* } @"$sSo6ParentO11InlineChildO027__CxxTemplateInstN6Parent11b7Child21e2InbC4IcEEV4testE6stringSSvg"()
+
+extension Parent.InInlineChild {
+  func doSomething() {
+  }
+}
+// CHECK: define hidden swiftcc void @"$sSo6ParentO11InlineChildO02InbC0V4testE11doSomethingyyF"()
+
+extension Parent.InSecondInlineChild {
+  var x: Int {
+    return 2
+  }
+}
+// CHECK: define hidden swiftcc i64 @"$sSo6ParentO11InlineChildO06SecondbC0O02IndbC0V4testE1xSivg"()
+
+extension Parent.InlineChild.InSecondInlineChild {
+  var y: Int {
+    return 3
+  }
+}
+// define hidden swiftcc i64 @"$sSo6ParentO11InlineChildO06SecondbC0O02IndbC0V4testE1ySivg"()
+
+// CHECK: define hidden swiftcc { i64, %swift.bridge* } @"$s4test3useySSSo6ParentO11InlineChildO027__CxxTemplateInstN6Parent11d7Child21g2IndE4IcEEVF"()
+// CHECK: call swiftcc { i64, %swift.bridge* } @"$sSo6ParentO11InlineChildO027__CxxTemplateInstN6Parent11b7Child21e2InbC4IcEEV4testE6stringSSvg"
+func use(_ x: Parent.TypedefInInlineChild) -> String {
+  let s = x.string
+  return s
+}
+
+// CHECK: define hidden swiftcc { i64, %swift.bridge* } @"$s4test4use2ySSSo6ParentO11InlineChildO027__CxxTemplateInstN6Parent11d7Child21g2IndE4IcEEVF"()
+// CHECK: call swiftcc { i64, %swift.bridge* } @"$sSo6ParentO11InlineChildO027__CxxTemplateInstN6Parent11b7Child21e2InbC4IcEEV4testE6stringSSvg"
+func use2(_ x: Parent.InlineChild.TypedefInInlineChild) -> String {
+  let s = x.string
+  return s
+}
+
+// define swiftcc void @"$s4testAAyySo6ParentO11InlineChildO02IncD0VF"()
+// CHECK: alloca %TSo6ParentO11InlineChildO027__CxxTemplateInstN6Parent11b7Child21e2InbC4IcEEV
+// CHECK: call %"struct.Parent::InlineChild::TemplateInInlineChild"* @_ZN6Parent11InlineChild21TemplateInInlineChildIcEC1Ev
+// CHECK: call swiftcc void @"$sSo6ParentO11InlineChildO02InbC0V4testE11doSomethingyyF"(
+// CHECK: call swiftcc i64 @"$sSo6ParentO11InlineChildO06SecondbC0O02IndbC0V4testE1xSivg"(
```

Comment:
Unfortunately, these checks aren't going to be super portable. On some platforms this will be an `i32`. You can either use regex or maybe convert to an execution test?